packages/block-library/src/navigation-link/edit.js

```diff
@@ -129,6 +131,43 @@ function NavigationLinkEdit( {
 		selection.addRange( range );
 	}
 
+	function handleCreateEntity( type, entityTitle ) {
+		return saveEntityRecord( 'postType', type, {
+			title: entityTitle,
+			status: 'publish',
+		} ).then( ( entity ) => {
+			// `entity` may not reject the Promise
+			// but may still be invalid. Here we
+			// tests for unexpected values and throw accordingly.
+			if ( null === entity || undefined === entity ) {
+				throw new TypeError(
+					'API response returned invalid entity.',
+					entity
+				);
+			}
+
+			const requiredEntityProps = [ 'id', 'title', 'link' ];
+
+			const entityMissingProperty = requiredEntityProps.find(
+				( entityProp ) => ! entity.hasOwnProperty( entityProp )
+			);
```

Comment:
When would this happen?

Reply:
I felt we were relying on `saveEntityRecord` always returning `entity` with the expected shape. If this ever changed the component would break without warning. This simple check adds a guard and ensures the error is handled gracefully in the UI. Moreover, it affords an obvious point for debugging.

That said, I totally understand you may feel this is unnecessary.

Reply:
@aduth I'm going to leave this in unless you object?

Reply:
I understand a desire to try to preempt potential bugs, but as a consumer we shouldn't need to feel in a position to guard against unforeseen breaking changes in all of the APIs we consume from. If we can, we should want to know with certainty the types of values we're expecting to be working with, and consequently avoid any code which does not align to that understanding.

My recommendation would be to remove this if it's something where we should be safe to know that each of these properties exists. I _think_ that it is, though the documentation of `saveEntityRecord` could definitely be improved in this regard.

The same note can be said about the `null === page` condition above. I see no scenario where `saveEntityRecord` could possibly resolve to a `null` value. I can see that it may be `undefined`. A simpler `! post` could suffice as well.